![Buffalo Soccer Header](https://carload.com/g/BuffaloSoccerWide400.jpg)

# Part 1b, Merge the OpenLiga data

After creating Pandas pickle files for each of the midweek leagues of interest, the German Cup (dfb Pokal), Europa League, and Champions League, It's time to merge them into a single dataframe. 

In [1]:
import pandas as pd

In [15]:
dfEuropa2021 = pd.read_pickle('data/buffalo/europa_2021.pkl')
dfEuropa2022 = pd.read_pickle('data/buffalo/europa_2022.pkl')
dfEuropa2023 = pd.read_pickle('data/buffalo/europa_2023.pkl')
dfChamps2021 = pd.read_pickle('data/buffalo/champions_2021.pkl')
dfChamps2022 = pd.read_pickle('data/buffalo/champions_2022.pkl')
dfChamps2023 = pd.read_pickle('data/buffalo/champions_2023.pkl')
dfPokal2021 = pd.read_pickle('data/buffalo/dfb_2021.pkl')
dfPokal2022 = pd.read_pickle('data/buffalo/dfb_2022.pkl')
dfPokal2023 = pd.read_pickle('data/buffalo/dfb_2023.pkl')

In [20]:
dfOpenLiga = pd.concat([dfEuropa2021, dfEuropa2022, dfEuropa2023,
                        dfChamps2021, dfChamps2022, dfChamps2023,
                        dfPokal2021, dfPokal2022, dfPokal2023])
dfOpenLiga.head()

matchID  ordDate team1Name    team2Name  result
0    62820   738133    Sparta      Brondby       3
1    62821   738133      Lyon      Rangers       1
2    62822   738133                              1
3    62823   738133                    PSV       3
4    62824   738133            mos spartak       0

As you can see, we don't care about very many details in these games. Partly because of flaws in the input data, the original idea of tracking every goal by minute proved to be unworkable. In fact, in the example above, we can't even tell who was playing in matchID 62822.

We'll save this into a single pickle file to merge with the master data.

In [22]:
dfOpenLiga.to_pickle('data/buffalo/openliga.pkl')

Because of errors in the data provided by OpenLiga, I had to create custom CSV files for the 2022 and 2023 Europa League seasons. Since they only include the matches played by Bundesliga teams, they were smaller than the full-roster 2021 file.

In [21]:
print (dfEuropa2021.shape)
print (dfEuropa2022.shape)
print (dfEuropa2023.shape)
print (dfOpenLiga.shape)

(61, 5)
(26, 5)
(21, 5)
(642, 5)
